File: 브루트 포스/2798/2798.cpp
#include <iostream>
#include <vector>
using namespace std;

int main()
{
   int n, m;
   int answer = 0;
   cin>>n>>m;
   vector<int> v(n);

   for (int i = 0; i < n; ++i)
      cin>>v[i];
   
   for (int i = 0; i < n - 2; ++i)
      for (int j = i + 1; j < n - 1; ++j)
         for (int k = j + 1; k < n; ++k)
         {
            int temp = v[i] + v[j] + v[k];
            if (answer < temp and temp <= m)
               answer = temp;
         }

   cout<<answer;

   return 0;
}

File: 브루트 포스/2798/2798.py
n, m = map(int, input().split())
arr = list(map(int, input().split()))
answer = 0

for i in range(n - 2):
   for j in range(i + 1, n - 1):
      for k in range(j + 1, n):
         temp = arr[i] + arr[j] + arr[k]
         if answer < temp and temp <= m:
            answer = temp
            
print(answer)